src/exchange/exchange.ts

```diff
@@ -17,72 +24,113 @@ export const devtoolsExchange: Exchange = ({ client, forward }) => {
       );
   }
 
+  // Expose graphql url for introspection
   window.__urql__ = {
     client,
     events: []
   };
 
-  // Initialize
-  sendToContentScript("init");
-
-  // Listen for messages
-  window.addEventListener("urql-in", e =>
-    handleMessage(client)((e as CustomEvent).detail)
-  );
+  // Listen for messages from content script
+  window.addEventListener("urql-in", (e: any) => {
+    const handler = (messageHandlers as any)[e.detail.type];
+    handler && handler(client)(e.detail);
+  });
+  sendToContentScript({ type: "init" });
 
   return ops$ => {
     return pipe(
       ops$,
+      map(addOperationContext),
       tap(handleOperation),
       forward,
+      map(addOperationResponseContext),
       tap(handleOperation)
     );
   };
 };
 
+const addOperationResponseContext = (op: OperationResult): OperationResult => {
+  return {
+    ...op,
+    operation: {
+      ...op.operation,
+      context: {
+        ...op.operation.context,
+        meta: {
+          ...op.operation.context.meta,
+          // @ts-ignore
```

Comment:
What's this for?

Reply:
This is because the urql types are only updated in 1.4.0 so these releases are quite co-dependent